run.sh

```diff
@@ -1,69 +1,95 @@
 #!/usr/bin/env bash
 BASE_DIR=$(cd $(dirname $0); pwd -P)
-CASE_FILE=$2
-#export RUST_LOG=debug
+CASE_FILE="cases.json"
+MODE=""
 
-run_boring_server() {
-  cargo run -- \
-  --client $BASE_DIR/../dist/Debug/bin/nss_bogo_shim \
-  --server $BASE_DIR/../boringssl/build/ssl/test/bssl_shim \
-  --rootdir $BASE_DIR/../boringssl/ssl/test/runner/ \
-  --test-cases $BASE_DIR/$CASE_FILE \
-  --client-writes-first
-}
+while [ $# -gt 0 ]; do
+    case $1 in
+        -v) export RUST_LOG=debug ;;
+        -m) MODE="$2"; shift ;;
+        -c) CASE_FILE="$2"; shift;;
+        *) echo "Error: Unknown argument."; exit 2 ;;
+    esac
+    shift
+done
 
-run_boring_client() {
-  cargo run -- \
-  --client $BASE_DIR/../boringssl/build/ssl/test/bssl_shim \
-  --server $BASE_DIR/../dist/Debug/bin/nss_bogo_shim \
-  --rootdir $BASE_DIR/../boringssl/ssl/test/runner/ \
-  --test-cases $BASE_DIR/$CASE_FILE
-}
+while [ $# -gt 0 ]; do
+    case $1 in
+        -v) export RUST_LOG=debug ;;
+        -m) MODE="$2"; shift ;;
+        -c) CASE_FILE="$2"; shift;;
+        *) echo "Error: Unknown argument."; exit 2 ;;
+    esac
+    shift
+done
 
-run_ossl_server() {
-  cargo run -- \
-  --client $BASE_DIR/../dist/Debug/bin/nss_bogo_shim \
-  --server $BASE_DIR/../openssl/test/ossl_shim/ossl_shim \
-  --rootdir $BASE_DIR/../boringssl/ssl/test/runner/ \
-  --test-cases $BASE_DIR/$CASE_FILE \
-  --client-writes-first \
-  --force-IPv4
-}
+CERT_DIR="$BASE_DIR/../boringssl/ssl/test/runner/"
+NSS_SHIM="$BASE_DIR/../dist/Debug/bin/nss_bogo_shim"
+BSSL_SHIM="$BASE_DIR/../boringssl/build/ssl/test/bssl_shim"
+OSSL_SHIM="$BASE_DIR/../openssl/test/ossl_shim/ossl_shim"
+
+SHIM_ARRAY=($NSS_SHIM $BSSL_SHIM $OSSL_SHIM)
+
+run_shim_pair() {
+  # If NSS acts as the client, interop needs this argument.
+  # It would become obsolete if bssl and ossl could actively initiate
+  # communication after the handshake.
+  CLIENT_WRITES=""
+  if [[ $1 = *"nss_bogo_shim"* ]]; then
```

Comment:
You could read the `$1` and `$2` into a `shim1` and `shim2` variable or something like that. Makes it easier to read.

Reply:
Done.